NServiceBus.Transport.AzureServiceBus 3.1.0 Updates (#5923)
Particular/docs.particular.net: transports/azure-service-bus/topology.md

#### Quotas and limitations

> The `ForwardingTopology` is subjected to [quotas](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quotas) of Azure Service Bus. A single topic can have up to 2000 subscriptions and 2000 filters/rules. For every logical endpoint one subscription is created that contains one rule per event type subscribed as shown on the topology overview.

Reviewer: ```suggestion
The `ForwardingTopology` is subject to [quotas](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quotas) of Azure Service Bus. A single topic supports up to 2000 subscriptions and 2000 filters/rules. For every logical endpoint, one subscription is created that contains one rule per event type subscribed, as shown on the topology overview.
```

As an example, in larger systems, there may be dozens of event types for various business components to interact with each other. Considering a system with 100 different event types, with every endpoint subscribing to 25% of the events, an endpoint would have to manage at least 25 rules. This means the `ForwardingTopology` can handle up to 80 different logical endpoints until it reaches its scaling limits. The more events there are and the more an endpoint is interested in subscribing to them, the fewer endpoints can be added to the system.

partial: hierarchy

> Since multiple rules per subscription are used, topics cannot be partitioned.

Reviewer 2: @danielmarbach Do you have a reference for this statement, like from the Microsoft docs?

Author: I'm not aware this is an existing restriction of Azure Service Bus. It seems we have added this as part of https://github.com/Particular/docs.particular.net/pull/4803/ six years ago. Maybe @SeanFeldman remembers what the context was. This statement need to be removed.

Reviewer 3: Partitioning has gone through a rollercoaster with ASB. It was introduced to the Premium tier, copying the Standard tier. Then, it was removed to be added back again. This sentence can be removed.

Reviewer 4: Partitioning has gone through a rollercoaster with ASB. It was introduced to the Premium tier, copying the Standard tier. Then, it was removed to be added back again. This sentence can be removed.

Reviewer 2: Thanks Sean and Daniel,
Our team ran a quick test as well and confirmed we are able to use partitioning in this scenario.

Author: https://github.com/Particular/docs.particular.net/commit/6b8dc689bf8cd38a251f49191f1e8d96b00da30c
